### PR TITLE
Updates to `1deg_jra55do_ryf`

### PR DESCRIPTION
**Pull request by dougiesquire**
This PR:
- Updates to a dev build of `access-om3` that uses FMS _without_ the `large_file` and `deprecated_io` flags set, and configures MOM to use netcdf4 - see https://github.com/COSIMA/access-om3/issues/188
- Moves to using environment modules to specify the exe in `config.yaml` - see https://github.com/COSIMA/access-om3/issues/168. Note the module name includes the `access-om3` commit hash.
- Renames the input directory from 'input' to 'INPUT' - see https://github.com/COSIMA/access-om3/issues/170
- Adds new Payu userscripts from `om3-scripts` - see https://github.com/COSIMA/access-om3/issues/182:
  - Replace local `archive/setup_cice_restarts.sh` with those in `om3-scripts`
  - Concatenate daily CICE output to monthly
  - Auto-generate Intake-ESM datastores

I'll try cherry-pick to other configs once this is merged

### CODE REVIEW

**Comment by anton-seaice** [comment]
We should update the wiki, and `OCN_modelio::` in `nuopc.runconfig` (Can we set all the lines starting with pio to unused?) (i.e. COSIMA/access-om3#185)

**Comment by anton-seaice** [comment]
I ran this as written:

```
ll -h archive/output000/access-om3.mom6.*
-rw-r----- 1 as2285 tm70 185M Jul 29 15:15 archive/output000/access-om3.mom6.h.native_1900_01.nc
-rw-r----- 1 as2285 tm70  80M Jul 29 15:15 archive/output000/access-om3.mom6.h.sfc_1900_01.nc
-rw-r----- 1 as2285 tm70 847K Jul 29 15:15 archive/output000/access-om3.mom6.h.static.nc
-rw-r----- 1 as2285 tm70  97M Jul 29 15:15 archive/output000/access-om3.mom6.h.z_1900_01.nc
```
with payu walltime of 1055s

and then ran with deflate=2:

```
-rw-r----- 1 as2285 tm70 211M Jul 29 15:35 archive/output001/access-om3.mom6.h.native_1900_02.nc
-rw-r----- 1 as2285 tm70  79M Jul 29 15:35 archive/output001/access-om3.mom6.h.sfc_1900_02.nc
-rw-r----- 1 as2285 tm70 993K Jul 29 15:35 archive/output001/access-om3.mom6.h.static.nc
-rw-r----- 1 as2285 tm70 111M Jul 29 15:35 archive/output001/access-om3.mom6.h.z_1900_02.nc
```

with payu walltime of 1022s

So it looks like the extra time for compression is maybe worth it. Noting tuning the IO layout might improve this.

**Comment by anton-seaice** [comment]
> I ran this as written:
> 
> ```
> ll -h archive/output000/access-om3.mom6.*
> -rw-r----- 1 as2285 tm70 185M Jul 29 15:15 archive/output000/access-om3.mom6.h.native_1900_01.nc
> -rw-r----- 1 as2285 tm70  80M Jul 29 15:15 archive/output000/access-om3.mom6.h.sfc_1900_01.nc
> -rw-r----- 1 as2285 tm70 847K Jul 29 15:15 archive/output000/access-om3.mom6.h.static.nc
> -rw-r----- 1 as2285 tm70  97M Jul 29 15:15 archive/output000/access-om3.mom6.h.z_1900_01.nc
> ```
> 
> with payu walltime of 1055s
> 
> and then ran with deflate=1:
> 
> ```
> -rw-r----- 1 as2285 tm70 211M Jul 29 15:35 archive/output001/access-om3.mom6.h.native_1900_02.nc
> -rw-r----- 1 as2285 tm70  79M Jul 29 15:35 archive/output001/access-om3.mom6.h.sfc_1900_02.nc
> -rw-r----- 1 as2285 tm70 993K Jul 29 15:35 archive/output001/access-om3.mom6.h.static.nc
> -rw-r----- 1 as2285 tm70 111M Jul 29 15:35 archive/output001/access-om3.mom6.h.z_1900_02.nc
> ```
> 
> with payu walltime of 1022s
> 
> So it looks like the extra time for compression is maybe worth it. Noting tuning the IO layout might improve this.

Oh february has 28 days and january has 31 ... the time difference might be unrelated to compression

**Comment by dougiesquire** [comment]
> Oh february has 28 days and january has 31 ... the time difference might be unrelated to compression

And the file size differences

**Review by dougiesquire — COMMENTED** [review]
~I don't think this will run will it?~ Scratch that, I misread your change. I assume you tried to run with these changes?

**Comment by anton-seaice** [comment]
Yes it works (or im having a 5 oclock moment!). It only removes lines for components which don't exist

**Comment by dougiesquire** [comment]
Let me know when you're happy and I'll do another test run with your added changes

**Comment by anton-seaice** [comment]
Nothing further @dougiesquire

**Comment by dougiesquire** [comment]
Thanks @anton-seaice. Test run ran successfully. But I'm going to bundle in a couple more changes to close https://github.com/COSIMA/access-om3/issues/168 and https://github.com/COSIMA/access-om3/issues/170 to reduce the amount of cherry-picking I have to do. I won't include changes for https://github.com/COSIMA/access-om3/issues/194 in this PR since this requires more testing.

**Comment by dougiesquire** [comment]
^ :eyes: @micaeljtoliveira

**Comment by micaeljtoliveira** [comment]
Oh no... We need to change the parsing of the comment to make sure we only trigger the workflow if the expression "cherry-pick" is at the beginning of the line.

**Comment by dougiesquire** [comment]
Maybe worth having a character at the start to distinguish the command? E.g. [Tommy's versioning commands](https://github.com/ACCESS-NRI/access-om2-configs/pull/129#issuecomment-2219695834) start with a "!".

**Comment by aekiss** [comment]
also trigger only for `!cherry-pick ` (with trailing space) or `!cherry-pick into `?

**Comment by micaeljtoliveira** [comment]
I think just adding the exclamation mark is enough to avoid 99.99% of unwanted triggers of the command. 

Triggering it when not specifying `into` is a good thing, as it gives some feedback to the user that might otherwise be left wondering why the comment didn't trigger the workflow.

**Comment by micaeljtoliveira** [comment]
See #92 for the fix.

**Comment by micaeljtoliveira** [comment]
It should now be safe to discuss cherry-picks in PRs without being interrupted by the github bot.

**Comment by aekiss** [comment]
cherry-pick cherry-pick cherry-pick cherry-pick cherry-pick cherry-pick cherry-pick cherry-pick

**Comment by aekiss** [comment]
seems to work :-)

**Comment by dougiesquire** [comment]
@anton-seaice, I've bundled a couple more things into this PR with the hope it might reduce the amount of cherry-picking and PRs. Let me know if you're happy and I'll do some history rewriting to simplify the cherry-picking.

File: input.nml [hunk 1]
```diff
@@ -20,3 +20,9 @@
     cf_compliance = .true.
 /
 
+&fms2_io_nml
+    ncchksz = 4194304
```
[the remark this answers is not in the document]
Comment (anton-seaice) [reply input.nml:24]: I still think this line could do with an interpretive note :)

**Review by anton-seaice — COMMENTED** [review]
Can you update the wiki please for where to set MOM IO & netcdf configs ?
I suggest we put this in four commits, and link to the related issue for each PR in the commit / description. 

Is there somewhere obvious to document it needs payu/dev or payu 1.1.4 to run ? (in config.yaml, in the README, somewhere else?)

I haven't tested, but expect its fine.

File: input.nml (hunk 1 above)
[the remark this answers is not in the document]
Comment (dougiesquire) [reply input.nml:24]: I think I'll avoid putting comments in a namelist group because it seems that what's allowed/not allowed is complicated

**Comment by anton-seaice** [comment]
This looks good:

- ncdump of MOM history and restart files shows format = netcdf-4 , shuffle = true, chunksize exist and are full size of x/y/time etc, deflate is 4. 
- payu finds the exe using the module, and the module name contains the access-om3 git commit used for the build. full binary path is in the manifests
- folder is named INPUT and no errors are given loading forcings / grids etc
- setup user scripts copies cice restart into work folder but there no extra copies in the output archive, intake esm datastore exists. We have however overwritten the monthly cice output with the daily cice output !

**Comment by anton-seaice** [comment]
Ill update the cice config and add to this PR

**Comment by anton-seaice** [comment]
For reasons that are unclear, the `hist_suffix` namelist config in cice appears to only support one character per history frequency. I can set this to 'd' and 'm' to progress the PR, or we can park it. Ill raise an issue tomorrow.

**Comment by anton-seaice** [comment]
This is blocked on https://github.com/COSIMA/access-om3/issues/201 for now

**Comment by anton-seaice** [comment]
I suggest we drop the userscripts from this PR and merge the rest :)

**Comment by dougiesquire** [comment]
We decided in a meeting today to set `hist_suffix` to:
- '.' for daily files
- 'm' for monthly files

The CICE filenames will look a bit yucky until https://github.com/COSIMA/access-om3/issues/201 is resolved, but we can then at least use the concatenation userscript without modification.

**Comment by anton-seaice** [comment]
>     * '.' for daily files

sadly i was wrong and it needs to be blank until we change the userscripts see https://github.com/COSIMA/om3-scripts/issues/21

**Review by anton-seaice — APPROVED** [review]
I've done the old change and approve trick :). This works now.

**Comment by dougiesquire** [comment]
> sadly i was wrong and it needs to be blank until we change the userscripts see https://github.com/COSIMA/om3-scripts/issues/21

I vote we change the userscript now to something that can work with both:
- the `*1day.mean*` format
- some workaround format for now

Is that possible? Then we won't end up with more dependent changes down the track

**Comment by anton-seaice** [comment]
> > sadly i was wrong and it needs to be blank until we change the userscripts see [COSIMA/om3-scripts#21](https://github.com/COSIMA/om3-scripts/issues/21)
> 
> I vote we change the userscript now to something that can work with both:
> 
>     * the `*1day.mean*` format
> 
>     * some workaround format for now
> 
> 
> Is that possible? Then we won't end up with more dependent changes down the track

Yep "should" be fine. We can do that change at any point going forward - it doesn't impact this PR. It just needs to be done before changing the filenames to the desired filenames.

**Comment by dougiesquire** [comment]
> it doesn't impact this PR

True - I wasn't thinking clearly sorry

**Comment by dougiesquire** [comment]
!cherry-pick 34e6e16 a8bd3a1 b191c68 81262fa 615c049 into 1deg_jra55do_iaf

**Comment by dougiesquire** [comment]
!cherry-pick d24a5d2c8f057ef3e7571fcde3abd6f2db13f753 2ee5b1d9e2852dcb8ab7a220e15d95eea09eceb6 e152ad48f0c9b41d9709b691cec1641275b6333a 9f551ec66734f5cd30c88f9e7479af4fdd000530 7beab257bfecb6d27227875dced02016f271a9ac into 1deg_jra55do_iaf

**Comment by aekiss** [comment]
@dougiesquire are you hitting merge conflicts due to the change in `MOM_input` format in `1deg_jra55do_ryf`?

Should I expedite https://github.com/COSIMA/access-om3/issues/200?

**Comment by dougiesquire** [comment]
There are conflicts in the `MOM_input`. But there are conflicts in every other modified file also, so expediting https://github.com/COSIMA/access-om3/issues/200 won't help much here.

**Pull request merged**